content/staging-area.md

```diff
@@ -161,61 +161,134 @@ We give two examples and the instructor can pick one or both:
-:alt: Staging basics
-:width: 100%
+**git commit** creates a new commit:
+```text
+
+  HEAD            .         project
+    |             .
+    |             .
+    |             .
+  HEAD~1          .
+    |             .
+    |             .
+    |             .
+  HEAD~2          .
+    |             .
+    |             .
+    |             .
+                  .
+  .git            .     working directory
+(history)         .      (files we see)
 ```
 
-Files can be untracked, modified, staged, or committed, and
-we have a variety of commands to go between states:
-
-```shell
-$ git add <path>       # stages all changes in file
-$ git add -p <path>    # stages while letting you choose which lines to take
-$ git commit           # commits the staged change
-$ git diff             # see **unstaged** changes
-$ git diff --staged    # see **staged** changes
-$ git rm               # removes a file
-$ git reset            # unstages staged changes
-                       # in latest Git: git restore --staged <path>
-$ git checkout <path>  # check out the latest staged version ( or committed
-                       # version if file has not been staged )
-                       # in latest Git: git restore <path>
+**git reset --soft HEAD~1** (move HEAD back by one but keep changes and stage
+them) would do the opposite of **git commit**.
+
+Going back to the last staged version:
+```text
+              git restore
+ [project*]  ------------>  project*
+
+
+
+  HEAD            .
+    |             .
+    |             .
+    |             .
+  HEAD~1          .
+    |             .
+    |             .
+    |             .
+                  .
+  .git            .     working directory
+(history)         .      (files we see)
 ```
 
-**Usage 1: you do various things at once**
-- You do various things at once, you want to commit them separately.
-- `git add` all the parts of each change.
-- `git diff` and `git diff --staged` to ensure you have all the parts.
-- `git commit`.
-- Repeat.
-
-**Usage 2: you want to checkpoint**
-- You want to save in-progress work before you make a final commit.
-- `git add` every change that improves the code.
-- `git checkout` every change that made things worse.
-- `git commit` as soon as you have created a nice self-contained unit (not too large, not too small).
-- Discuss/think about what is too large or too small.
-
-*Note: the "staging area" has also alternatively been referred to as
-the index and the cache*.
-
-
-## Example workflow
-
-```shell
-$ git add file.py                 # checkpoint 1
-$ git add file.py                 # checkpoint 2
-$ git add another_file.py         # checkpoint 3
-$ git add another_file.py         # checkpoint 4
-# ... further work on another_file.py ...
-$ git diff another_file.py        # diff w.r.t. checkpoint 4
-$ git checkout another_file.py    # oops go back to checkpoint 4
-$ git commit                      # commit everything that is staged
+Unstaging changes:
```

Comment:
```suggestion
Unstaging changes with **git restore --staged**:
```
Just a demo of what I think each diagram should be introduced with (really emphasize the command, don't only leave it hidden in the diagram)